Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp

```diff
@@ -191,6 +196,117 @@ uint16_t  *plTextFont::IInitFontTexture()
     DeleteDC( hDC );
     DeleteObject( hFont );
 
+    return data;
+#elif defined(HS_BUILD_FOR_MACOS)
+    int     nHeight, x, y, c;
+    char    myChar[ 2 ] = "x";
+    uint16_t  *tBits;
+
+    uint32_t       *bitmapBits;
+    uint32_t        bAlpha;
+
+    
+    // Figure out our texture size
+    if( fSize > 40 )
+        fTextureWidth = fTextureHeight = 1024;
+    else if( fSize > 20 )
+        fTextureWidth = fTextureHeight = 512;
+    else
+        fTextureWidth = fTextureHeight = 256;
+
+
+    // Create a new DC and bitmap that we can draw characters to
+    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceGenericRGB);
+    CGContextRef bitmapContext = CGBitmapContextCreate(NULL,
+                                                fTextureWidth,
+                                                fTextureHeight,
+                                                8,
+                                                4 * fTextureWidth,
+                                                colorSpace,
+                                                kCGImageAlphaPremultipliedLast);
+    CFRelease(colorSpace);
+    CGContextSetTextMatrix(bitmapContext, CGAffineTransformMakeScale(1, -1));
+    CGContextTranslateCTM(bitmapContext, 0, fTextureHeight);
+    CGContextScaleCTM(bitmapContext, 1, -1);
+    
+    bitmapBits = (uint32_t *)CGBitmapContextGetData(bitmapContext);
+
+    nHeight = (fSize * 192.0f) / 72.0f;
```

Comment:
I remembered I hardcoded a DPI of 192. I probably should not do that.

- [ ] Fix DPI check

Reply:
I'm going to leave this at what it's been hardcoded to for now. I think better integration could be done once the actual Mac client is in place. I need to really pass the display device through the pipeline - which the pipeline can do. Just not without the Mac client in place.